### DISCUSSION TITLE
Is it really necessary to use transpilePackages for internal packages?

### DISCUSSION THREAD

**Discussion by FelHa** [vercel/turborepo#7685] [Help] [answered]
### Summary

In the official documentation regarding internal packages, it states that you need to transpile them in the `next.config.js`: `transpilePackages: ['math-helpers']` ([sharing-code](https://turbo.build/repo/docs/handbook/sharing-code/internal-packages)).

Also, in the current starter examples (`npx create-turbo@latest`), `@repo/ui` is transpiled in the next apps.

However, all next apps work even without transpiling internal packages.

Do you have to transpile internal packages in the next.config? If yes, why? Or is it not necessary? If not, why is it done in both the documentation and the examples?

Ideally, I would like to bundle internal packages in the regular `build`/`dev` step without transpiling them in advance.

### Additional information

I used `npm` as a package manager. I would expect, that internal packages are just resolved via symlinks and bundled into the apps.

**Comment by anthonyshew** [comment, marked as answer]
Answer about Next.js is here: https://github.com/vercel/turbo/issues/7058#issuecomment-1979769109
